tests/02_eradiate/01_unit/scenes/phase/test_blend.py

```diff
@@ -144,3 +186,65 @@ def test_blend_array(modes_all):
     # Kernel dict generation succeeds
     ctx = KernelDictContext()
     assert phase.kernel_dict(ctx).load()
+
+
+def test_blend_array_2_components(modes_all):
+    """
+    Test instantiation and kernel dict generation with an array of weights.
+    """
+    # Constructing using 1D arrays for weights succeeds
+    weight1 = np.array([1.0, 0.7, 0.5, 0.3, 0.0])
+    weight2 = np.array([0.0, 0.3, 0.5, 0.7, 1.0])
+    phase = BlendPhaseFunction(
+        components=[
+            {"type": "isotropic"},
+            {"type": "rayleigh"},
+        ],
+        weights=[weight1, weight2],
+        bbox=[[0, 0, 0], [1, 1, 1]],
+    )
+
+    # Kernel dict generation succeeds
+    ctx = KernelDictContext()
+    phase_dict = phase.kernel_dict(ctx)
+    assert phase_dict.load()
+
+    # weight file is correct
+    weight_values = read_binary_grid3d(phase.weight_file)
+    assert np.allclose(weight_values, weight2)
+
+
+def test_blend_array_3_components(modes_all):
+    """
+    Test instantiation and kernel dict generation with an array of weights.
+    """
+    # Constructing using 1D arrays for weights succeeds
+    weight1 = np.array([1.0, 0.6, 0.2, 0.2, 0.0])
+    weight2 = np.array([0.0, 0.2, 0.6, 0.2, 0.5])
+    weight3 = np.array([0.0, 0.2, 0.2, 0.6, 0.5])
+    phase = BlendPhaseFunction(
+        components=[
+            {"type": "rayleigh"},
+            {"type": "isotropic"},
+            {"type": "isotropic"},
+        ],
+        weights=[weight1, weight2, weight3],
+        bbox=[[0, 0, 0], [1, 1, 1]],
+    )
+
+    # Kernel dict generation succeeds
+    ctx = KernelDictContext()
+    phase_dict = phase.kernel_dict(ctx)
+    assert phase_dict.load()
```

Comment:
I'd like to verify the type of the created object:
```suggestion
    assert phase_dict.load() is mi.PhaseFunction
```